### PR TITLE
Add simple_chip type to soup

### PR DESCRIPTION
**Pull request by abhijitxy**
- [x] Introduce simple_chip type into [tscircuit/soup](https://github.com/tscircuit/soup)

### CODE REVIEW

**Review by seveibar — CHANGES_REQUESTED** [review]
Lets keep the old one but mark it deprecated, sorry should have put this in the issue description hehe. For zod you can use the .describe("@deprecated")

**Comment by abhijitxy** [comment]
I've updated the code to keep the old source_simple_bug deprecated

File: src/source/index.ts [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
 export * from "./source_simple_capacitor"
 export * from "./source_simple_resistor"
 export * from "./source_simple_diode"
 export * from "./source_simple_ground"
-export * from "./source_simple_bug"
+
+/**
+ * @deprecated Use source_simple_chip instead. This will be removed in a future version.
+ */
+export const old_source_simple_bug = z.object({
```
Comment (seveibar) [inline src/source/index.ts:12]: We should keep the name the same, we want to make ot so anyone importing from @tscircuit/soup or @tscircuit/props doesnt have their code broken

File: src/any_soup_element.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export const any_soup_element = z.union([
   src.source_led,
   src.source_net,
   src.source_group,
-  src.source_simple_bug,
+  src.source_simple_chip,
```
Comment (seveibar) [inline src/any_soup_element.ts:20]: We still want to include the old bug component here

**Review by seveibar — CHANGES_REQUESTED** [review]
We want to avoid breaking compatibility with any libraries that use soup, so we have to keep exporting the same stuff, just add chip as an additional export

File: src/any_soup_element.ts [hunk 3]
```diff
@@ -17,7 +17,8 @@ export const any_soup_element = z.union([
   src.source_led,
   src.source_net,
   src.source_group,
-  src.source_simple_bug,
+  src.source_simple_bug.describe('@deprecated'),
```
Comment (seveibar) [inline src/any_soup_element.ts:20]:
```suggestion
  src.source_simple_bug,
```

Comment (seveibar) [reply src/any_soup_element.ts:20]: (you only need it in one place)

File: src/source/any_source_component.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { source_simple_resistor } from "./source_simple_resistor"
 import { source_simple_capacitor } from "./source_simple_capacitor"
 import { source_simple_diode } from "./source_simple_diode"
 import { source_simple_ground } from "./source_simple_ground"
-import { source_simple_bug } from "./source_simple_bug"
+import { source_simple_chip } from "./source_simple_chip"
```
Comment (seveibar) [inline src/source/any_source_component.ts:6]:
```suggestion
import { source_simple_bug } from "./source_simple_bug"
import { source_simple_chip } from "./source_simple_chip"
```

File: src/source/index.ts [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
+ * @deprecated Use source_simple_chip instead. This will be removed in a future version.
+ */
+export const source_simple_bug = z.object({
+  type: z.literal("source_component"),
+  ftype: z.string().optional(),
+  source_component_id: z.string(),
+  name: z.string(),
+  manufacturer_part_number: z.string().optional(),
+  supplier_part_numbers: z.string().optional(),
+}).describe('@deprecated');
```
Comment (seveibar) [inline src/source/index.ts:29]: somehow this file was reformatted (it has semicolons now)

most repos have a formatbot to fix this kind of issue, maybe i'll add it quickly

Comment (seveibar) [inline src/source/index.ts:29]: we shouldn't add definitions directly to a barrel export

```suggestion
export * from "./source_simple_bug
```

**Review by seveibar — CHANGES_REQUESTED** [review]
couple of things

**Review by seveibar — APPROVED** [review]
nice

**Pull request merged**